include/aws/http/proxy_strategy.h

```diff
@@ -399,6 +385,24 @@ struct aws_http_proxy_strategy_factory *aws_http_proxy_strategy_factory_new_tunn
     struct aws_allocator *allocator,
     struct aws_http_proxy_strategy_factory_tunneling_adaptive_ntlm_options *config);
 
+/**
+ * This is an experimental API.
+ *
+ * Constructor for callback functions
+ *
+ * @param callback function for sending user data to user, example - NTLM chalenge 
+ * @param callback function for getting user data, example - NTLM Cred,NTLM Response, Kerberos Token
+ * @return NULL
+ */
+
+AWS_HTTP_API
+int aws_http_proxy_connection_configure_callback(
+    aws_http_proxy_send_user_data_callback_fn func_1,
```

Comment:
These should be controlled by the config structure.

Reply:
Add factory config structure for creating callback configuration